src/test/java/org/cactoos/http/io/ReadBytes.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2018 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.http.io;
+
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+import org.cactoos.Bytes;
+import org.cactoos.Input;
+import org.cactoos.Scalar;
+import org.cactoos.scalar.StickyScalar;
+
+/**
+ * Reads available data from {@link Input} as {@link Bytes} only once.
+ *
+ * <p>There is no thread-safety guarantee.
+ *
+ * @since 0.1
+ */
+public final class ReadBytes implements Bytes {
```

Comment:
@victornoel Couldn't [BytesOf](https://github.com/yegor256/cactoos/blob/master/src/main/java/org/cactoos/io/BytesOf.java) from cactoos be used instead of this new class?

Reply:
@vzurauskas nope, that was the purpose of the todo solved here: `BytesOf` has the problem of closing the stream after reading it. I will update the javadoc and add a test to enforce this behaviour!